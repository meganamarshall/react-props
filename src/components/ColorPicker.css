.ColorPicker {
  button {
    margin: 5px;
  }
  
  .red {
    background-color: red;
  }

  .green {
    background-color: green;
  }

  .blue {
    background-color: blue;
  }
}
