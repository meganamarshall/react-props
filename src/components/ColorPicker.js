import React from 'react';
import styles from './ColorPicker.css';

export default function ColorPicker() {
  const redHandler = () => {
    console.log('red');
  };

  const greenHandler = () => {
    console.log('green');
  };

  const blueHandler = () => {
    console.log('blue');
  };

  return (
    <section className={styles.ColorPicker}>
      <button onClick={redHandler} className={styles.red}>Red</button>
      <button onClick={greenHandler} className={styles.green}>Green</button>  
      <button onClick={blueHandler} className={styles.blue}>Blue</button>
    </section>
  );
}
