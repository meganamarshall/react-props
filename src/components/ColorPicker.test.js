import React from 'react';
import ColorPicker from './ColorPicker';
import { shallow } from 'enzyme';


describe('ColorPicker component', () => {
  it('matches the colorpicker snapshot', () => {
    const wrapper = shallow(<ColorPicker />);
    expect(wrapper).toMatchSnapshot();
  });

  it('handles the red button', () => {
    console.log = jest.fn();
    const wrapper = shallow(<ColorPicker />);

    wrapper.find('button').at(0).simulate('click');
    expect(console.log).toHaveBeenCalledWith('red');
  });
});
