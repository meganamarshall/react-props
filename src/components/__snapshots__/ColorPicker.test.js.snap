// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker component matches the colorpicker snapshot 1`] = `
<section
  className="ColorPicker"
>
  <button
    className="red"
    onClick={[Function]}
  >
    Red
  </button>
  <button
    className="green"
    onClick={[Function]}
  >
    Green
  </button>
  <button
    className="blue"
    onClick={[Function]}
  >
    Blue
  </button>
</section>
`;
